validation/parser/face_vector_parser.py

```diff
@@ -1,10 +1,12 @@
+from functools import partial
+from parser.container_parser import ContainerParser
+from parser.pair import Pair
+from parser.parser_base import ParserBase
+from parser.pipeline.parser_pipeline import ParserPipeline
```

Comment:
NB: the `apply` method in this class is not a great name since it doesn't provide information. Might as well call it `__call__` so that then you can execute it as `pipeline()` since essentially the pipeline just is a [reduce](https://docs.python.org/3/library/functools.html#functools.reduce) aka `foldl` step.